src/python_testing/TC_AVSM_2_5.py

```diff
@@ -0,0 +1,232 @@
+        return "[TC-AVSM-2.5] Validate Audio Stream Allocation functionality with Server as DUT"
+
+    def pics_TC_AVSM_2_5(self):
+        return ["AVSM.S"]
+
+    def steps_TC_AVSM_2_5(self) -> list[TestStep]:
+        return [
+            TestStep("precondition", "Commissioning, already done", is_commissioning=True),
+            TestStep(
+                1, "TH reads FeatureMap attribute from CameraAVStreamManagement Cluster on TH_SERVER", "Verify ADO is supported."
+            ),
+            TestStep(
+                2,
+                "TH reads AllocatedAudioStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated audio streams in the list is 0.",
+            ),
+            TestStep(
+                3,
+                "TH reads MicrophoneCapabilities attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aMicrophoneCapabilities.",
+            ),
+            TestStep(
+                4,
+                "TH reads RankedVideoStreamPrioritiesList attribute from CameraAVStreamManagement Cluster on TH_SERVER.",
+                "Store this value in aRankedStreamPriorities.",
+            ),
+            TestStep(
+                5,
+                "TH sends the AudioStreamAllocate command with valid values of AudioCodec, ChannelCount, SampleRate and BitDepth from aMicrophoneCapabilities and StreamUsage from aRankedStreamPriorities.",
+                "DUT responds with AudioStreamAllocateResponse command with a valid AudioStreamID.",
+            ),
+            TestStep(
+                6,
+                "TH reads AllocatedAudioStreams attribute from CameraAVStreamManagement Cluster on TH_SERVER",
+                "Verify the number of allocated audio streams in the list is 1.",
+            ),
+            TestStep(
+                7,
+                "TH sends the AudioStreamAllocate command with values from step 4 except with ChannelCount set to 16(outside of valid range)",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+            TestStep(
+                8,
+                "TH sends the AudioStreamAllocate command with values from step 4 except with BitDepth set to 48(outside of valid range)",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+            TestStep(
+                9,
+                "TH sends the AudioStreamAllocate command with values from step 4 except with Samplerate set to 0(outside of valid range)",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+            TestStep(
+                10,
+                "TH sends the AudioStreamAllocate command with values from step 4 except with BitRate set to 0(outside of valid range)",
+                "DUT responds with a CONSTRAINT_ERROR status code.",
+            ),
+        ]
+
+    @async_test_body
```

Comment:
```suggestion
    @run_if_endpoint_matches(has_feature(Clusters.CameraAvStreamManagement,                                         Clusters.CameraAvStreamManagement.Bitmaps.Feature.kAudio))
```

Reply:
Thank you for your comments. We fixed it with our new commit.